src/main/java/dev/jbang/ScriptResource.java

```diff
@@ -43,19 +53,49 @@ public Path fetchIfNeeded(String resource) {
 	public Path fetchIfNeeded(String resource, String originalResource) {
 		if (Util.isURL(resource) || Util.isURL(originalResource)) {
 			try {
-				URI includeContext = new URI(originalResource);
-				URI thingToFetch = includeContext.resolve(resource);
+				URI thingToFetch = null;
+				if (Util.isURL(resource)) {
+					thingToFetch = new URI(resource);
+				} else {
+					URI includeContext = new URI(originalResource);
+					thingToFetch = includeContext.resolve(resource);
+				}
 				return Util.downloadAndCacheFile(thingToFetch.toString(), true);
 			} catch (URISyntaxException | IOException e) {
 				throw new IllegalStateException("Could not download " + resource + " relatively to " + originalResource,
 						e);
 			}
 		} else {
-			return file
-						.getAbsoluteFile()
-						.toPath()
-						.getParent()
-						.resolve(resource);
+			return originalSourceFileBaseDir
+											.getAbsoluteFile()
+											.toPath()
+											.resolve(resource);
 		}
 	}
+
+	private File setOriginalSourceFileBaseDir(String scriptURL) {
+		try (Scanner sc = new Scanner(new File(scriptURL))) {
+			while (sc.hasNextLine()) {
+				String line = sc.nextLine();
+				if (!line.trim().startsWith("package"))
+					continue;
+				String[] pkgLine = line.split("package");
+				if (pkgLine.length == 1)
+					continue;
+				String packageName = pkgLine[1];
+				packageName = packageName.split(";")[0].trim(); // remove ';'
+				String pkg = packageName.replace(".", File.separator); // a.b.c -> linux a/b/c, windows a\b\c
+				String pkgFile = pkg + File.separator + Util.getFileName(scriptURL);
+				return new File(scriptURL.substring(0, scriptURL.lastIndexOf(pkgFile)));
+			}
+		} catch (FileNotFoundException e) {
+			throw new RuntimeException(e);
```

Comment:
I would argue this methods must never fail. It can warn about some mismatched dir patterns but it should not fail hard.